mixnet/fisheryates.py

```diff
@@ -4,7 +4,11 @@
 
 class FisherYates:
     @staticmethod
-    def shuffle(elements: List, entropy: bytes) -> List:
+    def set_seed(seed: bytes) -> None:
+        random.seed(a=seed, version=2)
+
+    @staticmethod
+    def shuffle(elements: List) -> List:
```

Comment:
I think the old style is better. IIRC from the expects, the entropy changes over time. In this case we would set the whole system seed when calling `set_seed`, which is not what we want. It should be reset on every call to maintain integrity with other random calls. (This is python specific, but it actually fixates the intention that the entropy is just for the shuffling).

Reply:
Previously, yes, the entropy changes over time. But, at the recent [discussion](https://www.notion.so/Mixnet-Specification-807b624444a54a4b88afa1cc80e100c2?d=e35faabaf784477f9a7ea3e393ccb071&pvs=4#6a9d677c11b34ab7a3a00299c3c196e6), we've changed the design. The entropy is going to be injected less frequently, for example, 5 days. But, the topology update is going to be triggered more frequently, e.g. 30 minutes. So, the seed, which was set when the entropy was injected, should be maintained over subsequent topology updates until a new entropy is injected.

Reply:
I think this is the topic that we need to make a sync in https://github.com/logos-co/nomos-specs/pull/51#discussion_r1457675581

Reply:
Yes. I think in this case there is a missunderstanding here. `set_seed` should be instead `set_entropy` then we store the entropy and do not set the seed inmidiately. Then we use the old code. Otherwise is gonna be impossible to make simulations for example.

Reply:
Yes. If you remove the `mixnet_epoch` concept. We can use the old code. Also, not setting the seed immediately looks reasonable (https://github.com/logos-co/nomos-specs/pull/51#discussion_r1457694044).